test/config.js

```diff
@@ -0,0 +1,114 @@
+import path from 'path';
+import test from 'ava';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+const fixtureBasePath = path.resolve('test', 'fixtures', 'config');
+
+test('should return config only from home-directory when global binary used and `.np-config-json` in home-directory exists', async t => {
+	const homedirStub = sinon.stub();
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir1'));
+	const pkgDirStub = sinon.stub();
+	pkgDirStub.onFirstCall().returns(path.resolve(fixtureBasePath, 'pkg-dir'));
+	pkgDirStub.onSecondCall().returns(path.resolve(fixtureBasePath, 'local1'));
+	pkgDirStub.onThirdCall().returns(path.resolve(fixtureBasePath, 'local2'));
+
+	let i = 0;
+	while (i < 3) {
+		const config = proxyquire('../source/config', {
+			'is-installed-globally': true,
+			'pkg-dir': async () => {
+				return pkgDirStub();
+			},
+			os: {
+				homedir: homedirStub
+			}
+		});
+		t.deepEqual(await config(), {source: 'homedir/.np-config.json'});
+		i++;
+	}
+});
+
+test('should return config only from home-directory when global binary used and `.np-config.js` in home-directory exists', async t => {
+	const homedirStub = sinon.stub();
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir2'));
+	const pkgDirStub = sinon.stub();
+	pkgDirStub.onFirstCall().returns(path.resolve(fixtureBasePath, 'pkg-dir'));
+	pkgDirStub.onSecondCall().returns(path.resolve(fixtureBasePath, 'local1'));
+	pkgDirStub.onThirdCall().returns(path.resolve(fixtureBasePath, 'local2'));
+
+	let i = 0;
+	while (i < 3) {
+		const config = proxyquire('../source/config', {
+			'is-installed-globally': true,
+			'pkg-dir': async () => {
+				return pkgDirStub();
+			},
+			os: {
+				homedir: homedirStub
+			}
+		});
+		t.deepEqual(await config(), {source: 'homedir/.np-config.js'});
+		i++;
+	}
+});
+
+test('should return config only from package-directory when local binary used and `package.json` in package-directory exists', async t => {
+	const homedirStub = sinon.stub();
+	homedirStub.onFirstCall().returns(path.resolve(fixtureBasePath, 'homedir1'));
+	homedirStub.onSecondCall().returns(path.resolve(fixtureBasePath, 'homedir2'));
+	let i = 0;
+	while (i < 2) {
```

Comment:
Some here. No matter what configuration files the home directory contains, the local is always used.